Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -673,6 +678,23 @@ theorem erase_zero (a : α) : erase a (0 : α →₀ M) = 0 := by
   classical rw [← support_eq_empty, support_erase, support_zero, erase_empty]
 #align finsupp.erase_zero Finsupp.erase_zero
 
+theorem erase_eq_update (f : α →₀ M) (a : α) : f.erase a = update f a 0 :=
+  letI := Classical.decEq α
+  ext fun _ => (Function.update_apply _ _ _ _).symm
+
+-- not `simp` as `erase_of_not_mem_support` can prove this
+theorem erase_idem (f : α →₀ M) (a : α) :
+    erase a (erase a f) = erase a f := by
+  rw [erase_eq_update, erase_eq_update, update_idem]
+
+@[simp] theorem update_erase (f : α →₀ M) (a : α) (b : M) :
+    update (erase a f) a b = update f a b := by
```

Comment:
```suggestion
@[simp] theorem update_erase_self (f : α →₀ M) (a : α) (b : M) :
    update (erase a f) a b = update f a b := by
```
and you should have
```
-- @[simp] maybe?
theorem update_erase_of_ne (f : α →₀ M) {a a : α} (ha : a ≠ a') (b : M) :
    update (erase a f) a' b = erase a (update f a' b) := by
```

Reply:
This is called `insert_erase` on `Finset`, so I think the current name is correct.

Reply:
`Finset` is a weird comparison. Surely you should rather look at `Function`? I'm looking at `Function.update_same` and `Function.update_noteq` (which are pretty bad names indeed).

Reply:
Actually, your claim is wrong. `insert_erase` would correspond to `update (erase a f) a (f a) = f`.

Reply:
Perhaps `update_erase_eq_update` to match the `erase_update_eq_erase` I suggest below?

Reply:
That would be good, yes.

Reply:
I've made that change